pandas/indexes/base.py

```diff
@@ -270,13 +257,13 @@ def __new__(cls, data=None, dtype=None, copy=False, name=None,
             if dtype is None:
                 inferred = lib.infer_dtype(subarr)
                 if inferred == 'integer':
-                    from .numeric import Int64Index, UInt64Index
-                    try:
-                        return Int64Index(subarr.astype('i8'), copy=copy,
-                                          name=name)
-                    except OverflowError:
-                        return UInt64Index(subarr.astype('u8'), copy=copy,
-                                           name=name)
+                    out = cls._convert_to_int_index(subarr, copy, name)
+
+                    if out is not None:
+                        return out
+
+                    return Index(subarr, copy=copy,
```

Comment:
is this a change here? IOW, above we will then try to convert to a ``Float64Index``.